core/src/main/scala/org/http4s/rho/AuthedRhoService.scala

```diff
@@ -0,0 +1,65 @@
+package org.http4s
+package rho
+
+import org.http4s.rho.bits.PathAST.TypedPath
+import org.http4s.AuthedService
+import org.log4s.getLogger
+import shapeless.{HNil, HList}
+
+/** Constructor class for defining routes
+  *
+  * The [[AuthedRhoService]] provides a convenient way to define routes in a style
+  * similar to scalatra etc by providing implicit conversions and an implicit
+  * [[CompileService]] inside the constructor.
+  *
+  * {{{
+  *   val srvc = new AuthedRhoService[U] {
+  *     POST / "foo" / pathVar[Int] +? param[String]("param") |>> { (req: Request, p1: Int, param: String) =>
+  *       val authInfo: U = getAuth(req)
+  *       ...
+  *     }
+  *   }
+  *
+  * }}}
+  *
+  * @tparam U authInfo type for this service.
+  * @param routes Routes to prepend before elements in the constructor.
+  *
+  */
+class AuthedRhoService[U](routes: Seq[RhoRoute[_ <: HList]] = Vector.empty)
```

Comment:
In fact I think, it would be very unpleasant (maybe even impossible) to do it without making it a DSL. You can't call getAuth inside a RhoService which you need to give to transform.